src/Webapi/Canvas/Webapi__Canvas__Canvas2d.re

```diff
@@ -78,131 +78,147 @@ type style(_) =
-      [@bs.as "no-repeat"] | `noRepeat
+      | [@as "repeat-x"] `repeatX
+      | [@as "repeat-y"] `repeatY
+      | [@as "no-repeat"] `noRepeat
     ]
-  )
-  => pattern = "";
+  ) =>
+  pattern;
 
 /* Paths */
-[@bs.send.pipe : t] external beginPath : unit = "";
-[@bs.send.pipe : t] external closePath : unit = "";
-[@bs.send.pipe : t] external fill : unit = "";
-[@bs.send.pipe : t] external stroke : unit = "";
-[@bs.send.pipe : t] external clip : unit = "";
-[@bs.send.pipe : t] external moveTo : (~x: float, ~y: float) => unit = "";
-[@bs.send.pipe : t] external lineTo : (~x: float, ~y: float) => unit = "";
-[@bs.send.pipe : t] external quadraticCurveTo : (~cp1x: float, ~cp1y: float, ~x: float, ~y: float) => unit = "";
-[@bs.send.pipe : t] external bezierCurveTo : (~cp1x: float, ~cp1y: float, ~cp2x: float, ~cp2y: float, ~x: float, ~y: float) => unit = "";
-[@bs.send.pipe : t] external arcTo : (~x1: float, ~y1: float, ~x2: float, ~y2: float, ~r: float) => unit = "";
-[@bs.send.pipe : t] external arc : (~x: float, ~y: float, ~r: float, ~startAngle: float, ~endAngle: float, ~anticw: bool) => unit = "";
-[@bs.send.pipe : t] external rect : (~x: float, ~y: float, ~w: float, ~h: float) => unit = "";
-[@bs.send.pipe : t] external isPointInPath : (~x: float, ~y: float) => bool = "";
+[@send] external beginPath: t => unit;
+[@send] external closePath: t => unit;
+[@send] external fill: t => unit;
+[@send] external stroke: t => unit;
+[@send] external clip: t => unit;
+[@send] external moveTo: (t, ~x: float, ~y: float) => unit;
+[@send] external lineTo: (t, ~x: float, ~y: float) => unit;
+[@send] external quadraticCurveTo: (t, ~cp1x: float, ~cp1y: float, ~x: float, ~y: float) => unit;
+[@send]
+external bezierCurveTo:
+  (t, ~cp1x: float, ~cp1y: float, ~cp2x: float, ~cp2y: float, ~x: float, ~y: float) => unit;
+[@send] external arcTo: (t, ~x1: float, ~y1: float, ~x2: float, ~y2: float, ~r: float) => unit;
+[@send]
+external arc:
+  (t, ~x: float, ~y: float, ~r: float, ~startAngle: float, ~endAngle: float, ~anticw: bool) => unit;
+[@send] external rect: (t, ~x: float, ~y: float, ~w: float, ~h: float) => unit;
+[@send] external isPointInPath: (t, ~x: float, ~y: float) => bool;
 
 /* Text */
-[@bs.set] external font : (t, string) => unit = "";
-[@bs.set] external textAlign : (t, string) => unit = "";
-[@bs.set] external textBaseline : (t, string) => unit = "";
-[@bs.send.pipe : t] external fillText : (string, ~x: float, ~y: float, ~maxWidth: float=?) => unit = "";
-[@bs.send.pipe : t] external strokeText : (string, ~x: float, ~y: float, ~maxWidth: float=?) => unit = "";
-[@bs.send.pipe : t] external measureText : string => measureText = "";
-[@bs.get] external width : measureText => float = "";
+[@set] external font: (t, string) => unit;
+[@set] external textAlign: (t, string) => unit;
+[@set] external textBaseline: (t, string) => unit;
+[@send]
+external fillText: (t, string, ~x: float, ~y: float, ~maxWidth: float=?, [@ignore] unit) => unit;
```

Comment:
It looked like it was, but I figured I'd add it explicitly to be clear this API works as intended (also in case that implicit and possibly undocumented behaviour changes).